File: apps/web/src/types/aiResearch.ts
export type AIAnalysisType =
  | "EVENT_SUMMARY"
  | "INSTRUMENT_IMPACT"
  | "MULTI_EVENT_SYNTHESIS"
  | "RESEARCH_QUESTION";

export interface AIProviderStatus {
  provider_key: string;
  model: string;
  model_name: string;
  selectable_models: string[];
  configured: boolean;
  available: boolean;
  mode:
    | "DISABLED"
    | "FAKE"
    | "REAL_CONFIGURED"
    | "REAL_AVAILABLE"
    | "REAL_UNAVAILABLE";
  real_provider_available: boolean;
  base_url_summary: string | null;
  last_success_at: string | null;
  last_failure_at: string | null;
  last_error_code: string | null;
  capabilities: string[];
  warnings: string[];
  message: string;
}

export interface AIProviderTestResult {
  success: boolean;
  provider_key: string;
  model_name: string;
  mode: AIProviderStatus["mode"];
  latency_ms: number | null;
  error_code: string | null;
  warnings: string[];
}

export interface ResearchEvidence {
  evidence_id: string;
  information_item_id: string | null;
  market_event_id: string | null;
  evidence_text: string;
  evidence_location: string | null;
}

export interface ResearchInsight {
  insight_id: string;
  analysis_run_id: string;
  insight_type: string;
  title: string;
  summary: string;
  impact_direction: string;
  importance_score: string;
  confidence: string;
  time_horizon: string | null;
  key_facts: string[];
  uncertainties: string[];
  research_questions: string[];
  structured_output: Record<string, unknown>;
  schema_version: number;
  created_at: string;
  evidence: ResearchEvidence[];
  label: string;
}

export interface AIAnalysisRun {
  analysis_id: string;
  provider_key: string;
  model_name: string;
  analysis_type: AIAnalysisType;
  prompt_template_key: string;
  prompt_version: string;
  input_document_ids: string[];
  input_event_ids: string[];
  instrument_ids: string[];
  user_question: string | null;
  status: "CREATED" | "RUNNING" | "COMPLETED" | "FAILED";
  input_token_count: number | null;
  output_token_count: number | null;
  total_token_count: number | null;
  estimated_cost: string | null;
  cost_currency: string | null;
  is_real_provider: boolean;
  started_at: string | null;
  completed_at: string | null;
  failed_at: string | null;
  error: { code: string; message: string } | null;
  correlation_id: string;
  created_at: string;
  replayed: boolean;
  insight: ResearchInsight | null;
  capabilities: Record<string, boolean>;
}

export interface AIAnalysisPage {
  items: AIAnalysisRun[];
  page: number;
  page_size: number;
  total: number;
}

export interface ResearchInsightPage {
  items: ResearchInsight[];
  page: number;
  page_size: number;
  total: number;
}

export interface AIAnalysisCreateBody {
  analysis_type: AIAnalysisType;
  event_ids: string[];
  information_item_ids: string[];
  instrument_ids: string[];
  question: string | null;
  idempotency_key: string;
}

export type AIResearchDepth = "FAST" | "STANDARD" | "DEEP";
export type AIResearchTaskStatus =
  | "CREATED"
  | "PREPARING_DATA"
  | "RUNNING_AGENTS"
  | "DEBATING"
  | "RISK_REVIEW"
  | "GENERATING_REPORT"
  | "COMPLETED"
  | "PARTIALLY_COMPLETED"
  | "FAILED"
  | "CANCELED";

export interface AIResearchAgentStep {
  step_id: string;
  role: string;
  role_label: string;
  ordinal: number;
  status: "PENDING" | "RUNNING" | "COMPLETED" | "FAILED" | "SKIPPED";
  title: string;
  summary: string;
  structured_output: Record<string, unknown>;
  citations: Array<Record<string, unknown>>;
  input_token_count: number | null;
  output_token_count: number | null;
  error: { code: string; message: string } | null;
  started_at: string | null;
  completed_at: string | null;
}

export interface AIResearchReportSummary {
  report_id: string;
  title: string;
  executive_summary: string;
  stance: string;
  confidence: string;
  schema_version: number;
  created_at: string;
}

export interface AIResearchWorkflowEvent {
  event_id: string;
  sequence: number;
  event_type: string;
  status: string;
  node_name: string;
  agent_role: string | null;
  tool_name: string | null;
  payload: Record<string, unknown>;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

export interface AIResearchArtifact {
  artifact_id: string;
  artifact_key: string;
  artifact_type: string;
  title: string;
  content_markdown: string;
  ordinal: number;
  metadata: Record<string, unknown>;
  source_ids: string[];
  created_at: string;
}

export interface AIResearchTask {
  task_id: string;
  instrument: { id: string; symbol: string; exchange: string; name: string };
  question: string;
  depth: AIResearchDepth;
  start_date: string;
  end_date: string;
  provider_key: string;
  model_name: string;
  engine_key: string;
  engine_version: string;
  checkpoint_key: string;
  execution_attempt: number;
  last_checkpoint_at: string | null;
  is_real_provider: boolean;
  status: AIResearchTaskStatus;
  progress_percent: number;
  current_stage: string;
  warnings: string[];
  error: { code: string; message: string } | null;
  correlation_id: string;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
  steps: AIResearchAgentStep[];
  events: AIResearchWorkflowEvent[];
  artifacts: AIResearchArtifact[];
  report: AIResearchReportSummary | null;
  capabilities: Record<string, boolean>;
}

export interface AIResearchTaskPage {
  items: AIResearchTask[];
  page: number;
  page_size: number;
  total: number;
}

export interface AIResearchReport extends AIResearchReportSummary {
  task_id: string;
  sections: Record<string, unknown>;
  citations: Array<Record<string, unknown>>;
  limitations: string[];
  markdown: string;
  disclaimer: string;
}

export interface AIResearchTaskCreateBody {
  instrument_id: string;
  model_name: string;
  question: string;
  depth: AIResearchDepth;
  start_date: string;
  end_date: string;
  idempotency_key: string;
}

File: apps/api/src/alphadesk_api/schemas/ai_research.py
"""A01 public API schemas with forbidden secret/trading fields."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class AIProviderStatusResponse(BaseModel):
    provider_key: str
    model: str
    model_name: str
    selectable_models: list[str]
    configured: bool
    available: bool
    mode: str
    real_provider_available: bool
    base_url_summary: str | None
    last_success_at: datetime | None
    last_failure_at: datetime | None
    last_error_code: str | None
    capabilities: list[str]
    warnings: list[str]
    message: str


class AIProviderTestResponse(BaseModel):
    success: bool
    provider_key: str
    model_name: str
    mode: str
    latency_ms: int | None
    error_code: str | None
    warnings: list[str]


class AIProviderTestBody(BaseModel):
    """Select a configured model without allowing provider or secret overrides."""

    model_config = ConfigDict(extra="forbid")
    model_name: str | None = Field(default=None, min_length=1, max_length=128)


class AIAnalysisCreateBody(BaseModel):
    model_config = ConfigDict(extra="forbid")

    analysis_type: str
    event_ids: list[UUID] = Field(default_factory=list, max_length=50)
    information_item_ids: list[UUID] = Field(default_factory=list, max_length=50)
    instrument_ids: list[UUID] = Field(default_factory=list, max_length=100)
    question: str | None = Field(default=None, max_length=4000)
    idempotency_key: str = Field(min_length=1, max_length=128)


class ResearchEvidenceResponse(BaseModel):
    evidence_id: UUID
    information_item_id: UUID | None
    market_event_id: UUID | None
    evidence_text: str
    evidence_location: str | None


class ResearchInsightResponse(BaseModel):
    insight_id: UUID
    analysis_run_id: UUID
    insight_type: str
    title: str
    summary: str
    impact_direction: str
    importance_score: str
    confidence: str
    time_horizon: str | None
    key_facts: list[str]
    uncertainties: list[str]
    research_questions: list[str]
    structured_output: dict[str, object]
    schema_version: int
    created_at: datetime
    evidence: list[ResearchEvidenceResponse] = Field(default_factory=list)
    label: str = "AI生成，仅供研究参考。"  # noqa: RUF001 - product-required label


class AIAnalysisRunResponse(BaseModel):
    analysis_id: UUID
    provider_key: str
    model_name: str
    analysis_type: str
    prompt_template_key: str
    prompt_version: str
    input_document_ids: list[UUID]
    input_event_ids: list[UUID]
    instrument_ids: list[UUID]
    user_question: str | None
    status: str
    input_token_count: int | None
    output_token_count: int | None
    total_token_count: int | None
    estimated_cost: str | None
    cost_currency: str | None
    is_real_provider: bool
    started_at: datetime | None
    completed_at: datetime | None
    failed_at: datetime | None
    error: dict[str, str] | None
    correlation_id: UUID
    created_at: datetime
    replayed: bool = False
    insight: ResearchInsightResponse | None = None
    capabilities: dict[str, bool]


class AIAnalysisRunPageResponse(BaseModel):
    items: list[AIAnalysisRunResponse]
    page: int
    page_size: int
    total: int


class ResearchInsightPageResponse(BaseModel):
    items: list[ResearchInsightResponse]
    page: int
    page_size: int
    total: int


class AIIntegrityResponse(BaseModel):
    analysis_id: UUID
    valid: bool
    issues: list[dict[str, str]]
